src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,103 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    static final int DEFAULT_INITIAL_CAPACITY = 16;
+    static final int GROWTH_FACTOR = 2;
+    private int size;
+    private Node<K, V>[] map;
+
+    public MyHashMap() {
+        this.map = new Node[DEFAULT_INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int index = getIndex(key);
+        Node newNode = new Node<>(key == null ? 0 : key.hashCode(), key, value, null);
+        if (map[getIndex(key)] != null) {
```

Comment:
Too complex expression (I mean all nested if/else and loops) please introduce some methods to increase readability